<?xml version="1.0"?>

<launch>

<!-- Audibot with Twist disabled  --> 
  <include file="$(find ugv_course_launch)/launch/spawn_gazebo_robot.launch" >
    <arg name="robot" value="audibot" />
    <arg name="description_pkg" value="$(find audibot_description)" />
    <arg name="control_pkg" value="$(find audibot_control)" />
    <arg name="twist_control" value="false" />
    <arg name="start_x" value="-40.0" />
    <arg name="start_y" value="40.0" />
    <arg name="start_z" value="0.3" />
    <arg name="gps" value="true" />
    <arg name="gps_rate" value="10.0" />
    <arg name="pub_tf" value="0" />
    <arg name="use_sensors" value="1" />
    <arg name="pub_heading" value="0" />
    <arg name="tf_freq" value="1000.0" />
  </include>


<!-- Bicycle State Space Node -->
  <node pkg="bicycle_state_space" type="odom_node" name="odom_node" output="screen" >
    <param name="parent_frame" value="/map" />
    <param name="child_frame" value="/base_footprint" />
  </node>

<!-- Car Nav Node -->
  <node pkg="car_nav" type="car_nav_node" name="car_nav_node" output="screen" >
    <rosparam file="$(find car_nav)/yaml/costmap_params.yaml" ns="global_costmap" />
    <rosparam file="$(find car_nav)/yaml/global_planner_params.yaml" ns="global_planner" /> 
    <rosparam file="$(find car_nav)/yaml/local_costmap_params.yaml" ns="local_costmap" />
    <rosparam file="$(find car_nav)/yaml/teb_local_planner_params.yaml" ns="TebLocalPlannerROS" /> 
    
    <remap from="/move_base_simple/goal" to="/local_goal" />
  </node>
  
<!-- Set Nav Point Node -->
  <node pkg="set_nav_point" type="set_nav_point" name="set_nav_point" output="screen" />
  
<!-- lane_extract node (for parameter settings) 
  <node pkg="spot_detection" type="lane_extract" name="lane_extract" output="screen" >
    <param name="hough_threshold" value="25" />
    <param name="hough_min_line" value="100" />
    <param name="hough_max_line" value="30" />
    <param name="filename" value="$(find spot_detection)/image/template.png" />
  </node>
  -->
  
<!-- Find Parking Node 
  <node pkg="spot_detection" type="find_parking" name="find_parking" output="screen" >
    <param name="hough_threshold" value="200" />
    <param name="hough_min_line" value="150" />
    <param name="hough_max_line" value="10" />
    <param name="h_center" value="30" />
    <param name="h_width" value="10" />
  </node>
  -->
  
<!-- Gazebo World -->
  <include file="$(find ugv_course_gazebo)/launch/gazebo_world.launch" >
    <arg name="world_name" value="$(find car_nav)/world/sample3.world" />
    <arg name="headless" value="false" />
  </include>

<!-- Rviz Configuration  --> 
  <node pkg="rviz" type="rviz" name="rviz" args="-d $(find car_nav)/rviz/config3.rviz" respawn="true" />

</launch>
